File: JavaLibrary/src/main/java/com/flat/simplex/parser/logic/Field.java
package com.flat.simplex.parser.logic;

import com.flat.simplex.lexer.Token;

public class Field {
    public enum Type {
        Instance, Parameter, Local
    }

    private String name;
    private int argumentId;
    private Token tokenSource;
    private Type type;

    public Field(Token tokenSource, String name, Type type) {
        this.tokenSource = tokenSource;
        this.name = name;
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public int getArgumentId() {
        return argumentId;
    }

    public Token getTokenSource() {
        return tokenSource;
    }
}

File: SimplexApp/simplex/lang/parser/logic/Field.js
class Field {
    constructor(tokenSource, name, type) {
        this.tokenSource = tokenSource;
        this.name = name;
        this.type = type;
        this.argumentId = 0;
    }

    getName() {
        return this.name;
    }

    getArgumentId() {
        return this.argumentId;
    }

    getTokenSource() {
        return this.tokenSource;
    }
}
Field.Local = 'Local';
Field.Instance = 'Instance';
Field.Parameter = 'Parameter';

module.exports = Field;